squirrel.api/src/main/java/org/dice_research/squirrel/frontier/Frontier.java

```diff
@@ -31,17 +31,7 @@
     List<CrawleableUri> getNextUris();
 
     /**
-     * Add this URIs to the {@link Frontier}s internal queue if the internal
-     * rules of the {@link Frontier} allow it.
-     *
-     * @param uri
-     *            the URI that should be added to the {@link Frontier}
-     */
-    void addNewUri(CrawleableUri uri);
```

Comment:
The same as for the queue interface: please do not simply remove methods from such a central interface. You can mark it as deprecated and even add a default implementation which uses the addNewUris method.